opentelemetry-sdk/src/logs/in_memory_exporter.rs

```diff
@@ -157,14 +155,9 @@ impl InMemoryLogExporter {
     /// let emitted_logs = exporter.get_emitted_logs().unwrap();
     /// ```
     ///
-    pub fn get_emitted_logs(&self) -> LogResult<Vec<LogDataWithResource>> {
-        let logs_guard = self
-            .logs
-            .lock()
-            .map_err(|e| OTelSdkError::InternalFailure(format!("Failed to lock logs: {}", e)))?;
-        let resource_guard = self.resource.lock().map_err(|e| {
-            OTelSdkError::InternalFailure(format!("Failed to lock resource: {}", e))
-        })?;
+    pub fn get_emitted_logs(&self) -> Result<Vec<LogDataWithResource>, String> {
+        let logs_guard = self.logs.lock().map_err(|e| e.to_string())?;
```

Comment:
If we use `anyhow` and an error type we'd also be able to use `[From]` to get rid of the explicit  to-stringing here